compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1096,6 +1096,18 @@ private boolean isTupleTypeAssignable(BType source, BType target, Set<TypePair>
         return true;
     }
 
+    private boolean checkTupleMemberIsNoType(BType memberType) {
+        switch (memberType.tag) {
+            case TypeTags.NONE:
+                return true;
+            case TypeTags.TUPLE:
+                BTupleType tupleType = (BTupleType) memberType;
+                return tupleType.tupleTypes.stream().allMatch(this::checkTupleMemberIsNoType);
```

Comment:
We try to avoid using streams due to perf impact identified previously. Shall we replace this with a loop?